src/Front/DesugarPatTup.hs

```diff
@@ -0,0 +1,35 @@
+module Front.DesugarPatTup where
+
+import qualified Common.Compiler as Compiler
+import Common.Identifiers
+import qualified Front.Ast as A
+import           Data.Bifunctor                 ( first )
+
+desugarTupPat :: A.Program -> Compiler.Pass A.Program
+desugarTupPat (A.Program decls) = return $ A.Program $ desugarTop <$> decls
+    where desugarTop (A.TopDef defs) = A.TopDef $ desugarDef defs
```

Comment:
nit: can you move `desugarTop` etc. to the top-level, rather than sticking them all in this huge `where` block?

Reply:
Of course you'll need to add type annotations etc. but that should only serve to better document this code.